### PR TITLE
Avoid a race condition when assigning sequence numbers in the constructor of EzspFrameRequest.

### PR DESCRIPTION
**Pull request by sesuncedu**
See: #791 

Signed-off-by: Simon Spero <sesuncedu@gmail.com>

### CODE REVIEW

**Comment by sesuncedu** [comment]
Sigh: tests aren't running in a fresh VM.

File: com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ezsp/EzspFrameRequest.java [hunk 1]
```diff
@@ -40,16 +41,13 @@
...
 
     /**
      * Constructor used to create an outgoing frame
      */
     protected EzspFrameRequest() {
-        sequenceNumber = (int) sequence.getAndIncrement();
-        if (sequenceNumber == 254) {
-            sequence.set(1);
-        }
+        sequenceNumber = sequence.getAndUpdate(n -> n <254 ? n+1 : 1);
```
Comment (cdjackson) [inline com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ezsp/EzspFrameRequest.java]: Please use the old notation and synchronise the method for backward compatibility.

**Comment by cdjackson** [comment]
Did you mean to close this, or do you plan to provide this in a separate PR?

**Comment by sesuncedu** [comment]
Separate PR with the stupidity squashed.

Also, I am now not 100% sure that the frame request  is the right place to be assigning sequence numbers in the first place (if the requirement is that they be presented to the ncp in sequential order, then they should be assigned when they are transmitted, since the order in which they are presented may not be the same as the one in which they were created.

**Comment by cdjackson** [comment]
Fair point - feel free to change this to go into the ASH/SPI handlers?

**Pull request closed without merging**